components/02-molecules/banner/grand-hero/_yds-grand-hero.scss

```diff
@@ -0,0 +1,78 @@
+@use '../../../00-tokens/tokens';
+@use '../../../01-atoms/atoms';
+
+$break-grand-hero-banner: tokens.$break-m;
+
+.grand-hero-banner {
+  @include tokens.spacing-page-section($flush-top: true, $flush-bottom: true);
+
+  padding: 0;
+}
+
+.grand-hero-banner__content-wrapper {
+  position: relative;
+  overflow: hidden;
+  min-height: 30rem;
+  display: grid;
+  align-items: end;
+
+  @media (min-width: $break-grand-hero-banner) {
+    align-items: center;
+  }
+
+  [data-grand-hero-overlay-variation='contained'] & {
+    justify-content: start;
+    padding-bottom: var(--size-spacing-6);
+
+    @media (min-width: $break-grand-hero-banner) {
+      padding-bottom: 0;
+    }
+  }
+}
+
+.grand-hero-banner__image {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+
+  img {
+    object-fit: cover;
+    height: 100%;
+    width: 100%;
+  }
+}
+
+.grand-hero-banner__content {
+  position: relative;
+  opacity: 0.6;
+
+  // prettier-ignore
+  padding: var(--size-spacing-5) var(--size-spacing-7);
+
+  [data-grand-hero-overlay-variation='contained'] & {
+    @media (min-width: $break-grand-hero-banner) {
+      max-width: var(--size-component-layout-width-content);
+      margin: var(--size-spacing-site-gutter);
+    }
+  }
+
+  [data-grand-hero-overlay-variation='full'] & {
+    height: 100%;
+    display: flex;
+    align-items: center;
+  }
+}
+
+.grand-hero-banner__heading {
+  @include tokens.h2-yale-new;
+
+  margin-bottom: var(--size-spacing-2);
```

Comment:
The spacing is off between the text elements. In the designs, it looks like there's a consistent spacing of 9px between the heading, the text, and the link. Since we don't have a 9px variable, I'd suggest we use `--size-spacing-3` since it's equivalent to 8px.

We'll also need to remove the bottom margin from the last `p` tag inside the "snippet" element so that it doesn't create too much space. I'd suggest doing something like we did on the Callout: https://github.com/yalesites-org/component-library-twig/blob/develop/components/02-molecules/callout/_yds-callout.scss#L83-L89

And, since all of the content items are optional, I think using flex or grid (with the `gap` attribute) would be a better solution, so that we don't have dangling margins above, or below any of the items. In order for that to work, you'll need to delete the `__text` element so that everything's at the same level. But I think that's okay - I couldn't see any reason to keep that wrapper.